homeassistant/components/mqtt/config_integration.py

```diff
@@ -0,0 +1,197 @@
+"""Support for MQTT platform config setup."""
+from __future__ import annotations
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_CLIENT_ID,
+    CONF_DISCOVERY,
+    CONF_PASSWORD,
+    CONF_PORT,
+    CONF_PROTOCOL,
+    CONF_USERNAME,
+    Platform,
+)
+from homeassistant.helpers import config_validation as cv
+
+from . import (
+    alarm_control_panel as alarm_control_panel_platform,
+    binary_sensor as binary_sensor_platform,
+    button as button_platform,
+    camera as camera_platform,
+    climate as climate_platform,
+    cover as cover_platform,
+    device_tracker as device_tracker_platform,
+    fan as fan_platform,
+    humidifier as humidifier_platform,
+    light as light_platform,
+    lock as lock_platform,
+    number as number_platform,
+    scene as scene_platform,
+    select as select_platform,
+    sensor as sensor_platform,
+    siren as siren_platform,
+    switch as switch_platform,
+    vacuum as vacuum_platform,
+)
+from .const import (
+    ATTR_PAYLOAD,
+    ATTR_QOS,
+    ATTR_RETAIN,
+    ATTR_TOPIC,
+    CONF_BIRTH_MESSAGE,
+    CONF_BROKER,
+    CONF_CERTIFICATE,
+    CONF_CLIENT_CERT,
+    CONF_CLIENT_KEY,
+    CONF_DISCOVERY_PREFIX,
+    CONF_KEEPALIVE,
+    CONF_TLS_INSECURE,
+    CONF_TLS_VERSION,
+    CONF_WILL_MESSAGE,
+    DEFAULT_BIRTH,
+    DEFAULT_DISCOVERY,
+    DEFAULT_PREFIX,
+    DEFAULT_QOS,
+    DEFAULT_RETAIN,
+    DEFAULT_WILL,
+    PROTOCOL_31,
+    PROTOCOL_311,
+)
+from .util import _VALID_QOS_SCHEMA, valid_publish_topic
+
+DEFAULT_PORT = 1883
+DEFAULT_KEEPALIVE = 60
+DEFAULT_PROTOCOL = PROTOCOL_311
+DEFAULT_TLS_PROTOCOL = "auto"
+
+DEFAULT_VALUES = {
+    CONF_BIRTH_MESSAGE: DEFAULT_BIRTH,
+    CONF_DISCOVERY: DEFAULT_DISCOVERY,
+    CONF_PORT: DEFAULT_PORT,
+    CONF_TLS_VERSION: DEFAULT_TLS_PROTOCOL,
+    CONF_WILL_MESSAGE: DEFAULT_WILL,
+}
+
+PLATFORM_SCHEMAS_MODERN: dict[Platform, vol.Schema] = {
+    Platform.ALARM_CONTROL_PANEL: vol.All(
+        cv.ensure_list, [alarm_control_panel_platform.PLATFORM_SCHEMA_MODERN]  # type: ignore[has-type]
```

Comment:
MyPy is not capable of determining the type of `PLATFORM_SCHEMA_MODERN`, we need to ignore this to please MyPy.